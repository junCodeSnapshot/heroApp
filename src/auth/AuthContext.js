// Higer order component
import { createContext } from "react";

export const AuthContext = createContext();
