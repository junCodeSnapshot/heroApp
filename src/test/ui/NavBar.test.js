import React from 'react'
import { mount } from 'enzyme'
import '@testing-library/jest-dom'
import { MemoryRouter, Router } from 'react-router-dom'
import { AuthContext } from '../../auth/AuthContext'
import { Navbar } from '../../components/ui/Navbar'
import { types } from '../../types/types'
import { contextValueTrue } from '../fixtures/contextValue'

describe('Pruebas en <Navbar />', () => {

    const historyMock = {
        push: jest.fn(),
        replace: jest.fn(),
        location: {}, 
        listen: jest.fn(),
        createHref: jest.fn(),
    }

    const wrapper = mount(
        <AuthContext.Provider value={contextValueTrue}>
            <MemoryRouter>
                <Router history={historyMock}>
                    <Navbar />
                </Router>
            </MemoryRouter>
        </AuthContext.Provider>
    );

    afterEach(() => {
        jest.clearAllMocks();
    })

    test('debe de mostrarse correctamente ', () => {
      expect(wrapper).toMatchSnapshot();  
    });

    test('debe de llamar el logout y el usar history', () => {
        wrapper.find("button").prop('onClick')();

        expect(contextValueTrue.dispatch).toHaveBeenLastCalledWith({
            type: types.logout,
        });
        expect(historyMock.replace).toHaveBeenCalledWith('/login')
    })
    
        
})