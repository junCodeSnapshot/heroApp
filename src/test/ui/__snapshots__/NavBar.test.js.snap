// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Navbar /> debe de mostrarse correctamente  1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    onClick={[Function]}
  >
    Asociaciones
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Search Hero
      </a>
    </div>
  </div>
  <div>
    <ul
      className="navbar-nav ml-auto"
    >
      <span
        className="nav-item nav-link top-0 end-0 text-info"
      >
        Pedro
      </span>
      <button
        className="btn nav-item nav-link top-0 end-0"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
